src/main/java/undo/actions/Action.java

```diff
@@ -0,0 +1,43 @@
+package undo.actions;
+
+public interface Action<T> {
+
+    @SuppressWarnings("SpellCheckingInspection")
+    enum ActionType { CHANGELABELS }
+
+    ActionType getType();
+
+    String getDescription();
+
+    T act(T t);
+
+    T undo(T t);
+
+    // Returns an inverse of the Action.
+    @SuppressWarnings("unchecked")
+    default Action invert() {
+        Action that = this;
```

Comment:
We can use Action<T> here and above, then we don't need to SuppressWarnings or cast.